py/mkrules.cmake

```diff
@@ -69,7 +69,7 @@ add_custom_command(
 # it was last run, but it looks like it's not possible to specify that with cmake.
 add_custom_command(
     OUTPUT ${MICROPY_QSTRDEFS_LAST}
-    COMMAND ${Python3_EXECUTABLE} ${MICROPY_PY_DIR}/makeqstrdefs.py pp ${CMAKE_C_COMPILER} -E output ${MICROPY_GENHDR_DIR}/qstr.i.last cflags ${MICROPY_CPP_FLAGS} -DNO_QSTR sources ${MICROPY_SOURCE_QSTR}
+    COMMAND ${Python3_EXECUTABLE} ${MICROPY_PY_DIR}/makeqstrdefs.py pp ${CMAKE_C_COMPILER} -E ${MICROPY_CPP_FLAGS} -DNO_QSTR output ${MICROPY_GENHDR_DIR}/qstr.i.last cflags ${MICROPY_CPP_FLAGS} -DNO_QSTR sources ${MICROPY_SOURCE_QSTR}
```

Comment:
I think what's needed is to follow how `makeqstrdefs.py` is invoked in `py/mkrules.mk`, and add something like:
```
... cxxflags ${MICROPY_CPP_FLAGS} -DNO_QSTR ...
```
and if there are any additional C++ specific flags for the build, they also need to be added in that snippet.

Reply:
As need as I can tell, all the relevant include directories are added to `MICROPY_CPP_FLAGS` by your code at the bottom of `rp2`'s  `CMakeLists.txt` but unless I add `${MICROPY_CPP_FLAGS}` twice- following how the old `rp2` specific invocation of this command by adding it just after the `-E` flag.

From what I can understand these flags are being then passed *directly* to the C compiler command that `makeqsstrdefs.py` is invoking.

Reply:
I tried moving the duplicated args (after the -E flag) to after a new `cxxflags` entry, and the rp2 port compiled cleanly with `make USER_C_MODULES=../../examples/usercmodule`

I guess your problem is with compiling ulab though?

Reply:
`ulab` somehow seems to work fine, compiling against our house C++ modules is where it falls flat since they leverage the Pico SDK.

I'll give this a go since I haven't tried `cxxflags` yet, I was totally missing the distinction between `cxxflags` and `cflags`.

Reply:
I'm trying it now.  I seem to be able to build your entire repo of user-C-modules without issue, including ulab.

Reply:
Agreed, just tested it locally and force-pushed to see how Zephr reacts. Thank you!

Reply:
But to build with `pimoroni-pico/micropython/modules` I did need to make some changes like this:
```diff
--- a/micropython/modules/pico_display/pico_display.cpp
+++ b/micropython/modules/pico_display/pico_display.cpp
@@ -2,7 +2,7 @@
 #include "hardware/sync.h"
 #include "pico/binary_info.h"
 
-#include "../../../pimoroni-pico/libraries/pico_display/pico_display.hpp"
+#include "../../../libraries/pico_display/pico_display.hpp"
 
 using namespace pimoroni;
 
```

Reply:
Not the case here, which is weird- it *just works*- since I've updated the `continuous-integration` branch I'll bump our automated build and see if it still works.

Reply:
Seems fine (assuming you can see this) - https://github.com/pimoroni/pimoroni-pico/runs/2174755007

Need to fix the `ulab` `usermod.cmake` though since I left some debug output in the PR :grimacing: